src/editors/data/redux/thunkActions/video.js

```diff
@@ -232,31 +245,44 @@ export const deleteTranscript = ({ language }) => (dispatch, getState) => {
     language,
     videoId,
     onSuccess: () => {
-      const updateTranscripts = {};
-      Object.keys(transcripts).forEach((key) => {
-        if (key !== language) {
-          updateTranscripts[key] = transcripts[key];
-        }
-      });
-      dispatch(actions.video.updateField({ transcripts: updateTranscripts }));
+      const updatedTranscripts = transcripts.filter((code) => code !== language);
+      console.log({ updatedTranscripts });
+      dispatch(actions.video.updateField({ transcripts: updatedTranscripts }));
+    },
+  }));
+};
+
+export const updateTranscriptLanguage = ({ newLanguageCode, languageBeforeChange }) => (dispatch, getState) => {
+  const state = getState();
+  const { video: { transcripts, videoId } } = state;
+  selectors.video.getTranscriptDownloadUrl(state);
```

Comment:
> Is there a reason you are pulling out the video object out from the state instead of `const {transcripts, videoId} = state.video?

No there is no difference between the two.

why are you using selectors when you have the full state provided in the previous line?
Because the getTranscriptDownloadUrl is a selector which is dependent on state values, but itself not a state value. you're not wrong that I could make them all match, but that would require more mocking in the tests and I'm lazy